sdk/keyvault/azure-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/Poller.java

```diff
@@ -0,0 +1,425 @@
+
+package com.azure.security.keyvault.certificates;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.polling.PollResponse;
+import reactor.core.Disposable;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * This class offers API that simplifies the task of executing long-running operations against Azure service.
+ * The {@link Poller} consist of poll operation, cancel operation if supported by Azure service and polling interval.
+ * <p>
+ * It provides the following functionality:
+ *
+ * <ul>
+ *      <li>Querying the current state of long-running operations.</li>
+ *      <li>Requesting an asynchronous notification for long-running operation's state.</li>
+ *      <li>Cancelling the long-running operation if cancellation is supported by the service.</li>
+ *      <li>Triggering a poll operation manually.</li>
+ *      <li>Enable/Disable auto-polling.</li>
+ * </ul>
+ *
+ * <p><strong>Auto Polling</strong></p>
+ * Auto-polling is enabled by-default. It means that the {@link Poller} starts polling as soon as its instance is created. The {@link Poller} will transparently call the poll operation every polling cycle
+ * and track the state of the long-running operation. Azure services can return {@link PollResponse#getRetryAfter()} to override the {@code Poller.pollInterval} defined in the {@link Poller}.
+ * The {@link Poller#getStatus()} represents the status returned by the successful long-running operation at the time the last auto-polling or last manual polling, whichever happened most recently.
+ *
+ *<p><strong>Disable Auto Polling</strong></p>
+ * For those scenarios which require manual control of the polling cycle, disable auto-poling by calling {@code setAutoPollingEnabled#false} and perform manual poll
+ * by invoking {@link Poller#poll()} function. It will call poll operation once and update the {@link Poller} with the latest status.
+ * <p>When auto-polling is disabled, the {@link Poller} will not update its status or other information, unless manual polling is triggered by calling {@link Poller#poll()} function.
+ *
+ * <p>The {@link Poller} will stop polling when the long-running operation is complete or it is disabled. The polling is considered complete
+ * based on status defined in {@link PollResponse.OperationStatus}.
+ *
+ * <p><strong>Code Samples</strong></p>
+ *
+ * <p><strong>Instantiating and Subscribing to Poll Response</strong></p>
+ * {@codesnippet Poller.instantiationAndSubscribe}
+ *
+ * <p><strong>Wait/Block for Polling to complete</strong></p>
+ * {@codesnippet Poller.block}
+ *
+ * <p><strong>Disable auto polling and polling manually</strong></p>
+ * {@codesnippet Poller.poll}
+ *
+ * @param <T> Type of poll response value
+ * @see PollResponse
+ * @see PollResponse.OperationStatus
+ */
+public class Poller<T> {
```

Comment:
Should this be in azure-core module instead of KeyVault Certificates module?

Reply:
Yeah I have a PR out for it. 
https://github.com/Azure/azure-sdk-for-java/pull/5091
This is to ensure Certs API has the class until that PR gets merged.
Ignore reviewing this class here.